datafusion/core/src/datasource/memory.rs

```diff
@@ -241,6 +245,31 @@ impl TableProvider for MemTable {
                     )
                 })
                 .collect::<Result<Vec<_>>>()?;
+
+            // If there is a projection on the source, we also need to project orderings
```

Comment:
Instead of adjusting the ordering here, I wonder if it would be less code / "just work" if you applied the projection to `self.sort_order` first? the normal output properties calculation should work I think 🤔

Reply:
I don't think I can simply apply a straightforward projection to the `self.sort_order`, as they may still require special handling. For example, if we have an ordering `[a, b, c]` and the projection excludes column `a`, the remaining ordering `[b,c]` would not be valid. To avoid missing edge cases like this, I prefer consulting the equivalence API to ensure correctness.